### PR TITLE
refactor(connectors): use core day-window and timezone helpers

### LINKED ISSUE #2793: dedupe: timezone day-window math → use activityDayWindow from @remnic/core
ponytail-audit finding. Three hand-rolled copies of DST-aware day-window math while `@remnic/core` already exports robust `activityDayWindow` / `timezoneOffsetIso` / `assertValidTimezone` (canonical: packages/remnic-core/src/activity/digest.ts).

**Files**
- packages/connector-fireflies/src/normalize.ts (~lines 24–80)
- packages/connector-granola/src/normalize.ts (same helpers renamed granolaDayWindow)
- packages/connector-omi/src/normalize.ts (zonedDayBounds variant)
- packages/connector-reitti/src/client.ts (assertValidIanaTimezone duplicating core assertValidTimezone)

**Change**
- Import from `@remnic/core`; delete local copies. Keep thin field-name mapping where source payloads differ.

**Done when**
- `grep -rn "zonedDayStartIso\|zonedDayBounds\|assertValidIanaTimezone" packages/connector-*` returns zero.
- Connector normalize tests pass unchanged (fixtures prove identical output incl. a DST-boundary case).

### PR DESCRIPTION
**Pull request by joshuaswarren**
Fixes #2793

Three hand-rolled DST-aware day-window copies (fireflies zonedDayStartIso, granola granolaDayWindow, omi zonedDayBounds) and reitti's assertValidIanaTimezone duplicated @remnic/core's canonical activityDayWindow / timezoneOffsetIso / assertValidTimezone. Local copies deleted; imports from core; thin field-mapping only where payloads differ.

- `grep -rn 'zonedDayStartIso|zonedDayBounds|assertValidIanaTimezone' packages/connector-*` → zero
- fireflies 28/28, granola 27/27, omi 27/27, reitti 44/44; DST spring-forward fixtures byte-identical; one new omi DST-boundary test
- `check:pre-push` green; normalize.ts files shrank

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved timezone-aware activity filtering across Fireflies, Granola, and Omi, including daylight-saving transitions and skipped local dates.
  * Preserved UTC fallback behavior for invalid Omi timezones.
  * Standardized timezone validation for Reitti timeline and visit requests.

* **API Updates**
  * Added Omi day-window utilities while preserving legacy exports with deprecation warnings.
  * Added the Reitti timezone validation export while retaining the existing validation alias.
  * Consolidated timezone and day-window behavior for more consistent results across connectors.

### CODE REVIEW

File: packages/connector-omi/src/normalize.ts [hunk 1]
```diff
@@ -10,58 +10,47 @@
...
+export function omiDayWindow(
   date: string,
   timezone: string,
 ): { startIso: string; endIso: string } {
+  const { startUtc, endUtc } = activityDayWindow(date, timezone);
   return {
-    startIso: zonedDayStartIso(date, timezone),
-    endIso: zonedDayStartIso(nextIsoDate(date), timezone),
+    startIso: `${date}T00:00:00${timezoneOffsetIso(new Date(startUtc), timezone)}`,
+    endIso: `${nextIsoDate(date)}T00:00:00${timezoneOffsetIso(new Date(endUtc), timezone)}`,
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/connector-omi/src/normalize.ts]: Fixed in b21d74e68: `omiDayWindow` now renders each resolved instant's local wall-clock in the zone (`instantToOffsetIso`) instead of rebuilding `T00:00:00`, so skipped-midnight days keep core's instants. Added an `Africa/Cairo` 2026-04-24 (00:00→01:00) regression: bounds are `2026-04-24T01:00:00+03:00` / `2026-04-25T00:00:00+03:00` and parse exactly to `activityDayWindow`'s `startUtc`/`endUtc` (pre-fix start parsed 3,600,000 ms early, pulling in the prior local day's last hour); the prior-day end-bound truncation is asserted too.

File: packages/connector-omi/src/index.ts [hunk 2]
```diff
@@ -45,10 +45,9 @@ export {
   conversationToWearable,
   memoryToNativeMemory,
   nextIsoDate,
+  omiDayWindow,
   OMI_SOURCE_ID,
   timezoneOffsetIso,
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/connector-omi/src/index.ts:50]: Fixed in b21d74e68: `zonedDayBounds` and `zonedDayStartIso` are re-exported from the package entry point as `@deprecated` wrappers delegating to `omiDayWindow`. Entry-point import-compat test added (names resolve and match `omiDayWindow`).

File: packages/connector-reitti/src/index.ts [hunk 3]
```diff
@@ -39,7 +39,7 @@ export {
   REITTI_AUTH_MODES,
   REITTI_PLACE_TYPES,
   REITTI_TRANSPORT_MODES,
-  assertValidIanaTimezone,
+  assertReittiTimezone,
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/connector-reitti/src/index.ts:42]: Fixed in b21d74e68: `assertValidIanaTimezone` is re-exported as a `@deprecated` alias of `assertReittiTimezone`, with an entry-point import-compat test (valid zone passes, empty string → TypeError, unknown zone → RangeError). Also verified no caller in the repo depends on the old RangeError message text; throw types and conditions are identical to the previous local helper.

File: packages/connector-omi/src/normalize.ts [hunk 4]
```diff
@@ -10,58 +10,47 @@
...
-      timeZoneName: "longOffset",
-    }).formatToParts(instant);
-    const name = parts.find((part) => part.type === "timeZoneName")?.value ?? "GMT";
-    const match = name.match(/GMT([+-]\d{2}:\d{2})?/);
-    return match?.[1] ?? "+00:00";
+    return resolveTimezoneOffset(instant, timezone);
   } catch {
     return "+00:00";
   }
 }
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/connector-omi/src/normalize.ts:33]: Verified: a real difference existed. Pre-refactor `zonedDayBounds` never threw on an invalid zone (the offset helper's `+00:00` fallback), while the delegated path surfaced core's fail-fast assert — observed `RangeError: Invalid IANA timezone "Not/AZone"`. Fixed in 8e70fce6e via your first option: `omiDayWindow` normalizes an unknown zone to UTC before calling `activityDayWindow`, restoring the documented never-throw contract for both paths; regression test asserts `Not/AZone` yields the UTC day.

File: packages/connector-omi/src/normalize.ts [hunk 5]
```diff
@@ -10,61 +10,90 @@
...
+  const time = `${field("hour")}:${field("minute")}:${field("second")}`;
+  return `${day}T${time}${timezoneOffsetIso(instant, timezone)}`;
+}
+
+/**
+ * @deprecated Compatibility wrapper (pre-core-refactor export); delegates to
+ * `omiDayWindow`. Returns the resolved start bound, not a rebuilt midnight.
+ */
+export function zonedDayStartIso(date: string, timezone: string): string {
+  return omiDayWindow(date, timezone).startIso;
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/connector-omi/src/normalize.ts:97]: Fixed in aa77b7de6: `zonedDayBounds` and `zonedDayStartIso` each emit a once-per-process `process.emitWarning` DeprecationWarning pointing at `omiDayWindow` (codes `REMNIC_DEP_OMI_ZONEDDAYBOUNDS` / `REMNIC_DEP_OMI_ZONEDDAYSTARTISO`); a regression test asserts one warning per legacy name across repeated calls.

File: packages/connector-reitti/src/client.ts [hunk 6]
```diff
@@ -163,16 +164,23 @@ function isRecord(value: unknown): value is Record<string, unknown> {
...
-  }
+  assertValidTimezone(timezone);
+}
+
+/**
+ * @deprecated Alias of `assertReittiTimezone`, kept so pre-rename imports from
+ * this published package keep resolving. Same TypeError/RangeError behavior.
+ */
+export function assertValidIanaTimezone(timezone: string): void {
+  assertReittiTimezone(timezone);
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/connector-reitti/src/client.ts:184]: Fixed in aa77b7de6: `assertValidIanaTimezone` emits a once-per-process DeprecationWarning (`REMNIC_DEP_REITTI_ASSERT_VALID_IANA_TIMEZONE`) pointing at `assertReittiTimezone` before delegating; the compat test asserts exactly one warning across the valid, TypeError, and RangeError invocations.

File: packages/connector-reitti/src/index.test.ts [hunk 7]
```diff
@@ -218,3 +219,23 @@ test("verify rethrows a caller abort instead of reporting the provider unavailab
     (err: unknown) => err instanceof Error && err.name === "AbortError",
   );
 });
+
+test("legacy assertValidIanaTimezone still resolves and validates identically", async () => {
+  // Pre-rename consumers import assertValidIanaTimezone from the package
+  // entry; it must keep resolving with the same throw behavior and warn
+  // about the rename exactly once per process.
+  const warnings: Array<string | undefined> = [];
+  const onWarning = (warning: Error & { code?: string }) => warnings.push(warning.code);
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/connector-reitti/src/index.test.ts:228]: Declined — the finding is wrong against the tree. The cited range (`packages/connector-reitti/src/index.test.ts` 227-228) declares `onWarning` exactly once; `grep -c 'const onWarning'` on the file returns `1`, and the posted diff hunk itself shows a single declaration (the sibling `connector-omi/src/normalize.test.ts` test has the same one-declaration shape in a different file, which is the likely confusion). `tsc --noEmit` exits 0 for all four touched connector packages (omi, reitti, fireflies, granola) on head 6f3680d6a, so the file compiles. No duplicate exists to remove.

File: packages/remnic-core/src/activity/digest.ts [hunk 8]
```diff
@@ -135,10 +150,24 @@ export function activityDayWindow(date: string, timezone: string): { startUtc: s
...
+  // empty window that drops every record nominally of that day). A skipped
+  // date instead spans [transition instant, the next EXISTING date's start):
+  // it adopts the local day the calendar jumped into (Pacific/Apia
+  // 2011-12-30 adopts 2011-12-31's interval, sharing it). Windows stay
+  // non-degenerate, the last REAL day keeps its full window (its end is
+  // exactly the skipped date's start), and no date inherits the PREVIOUS
+  // date's interval.
+  while (endUtc <= startUtc) {
+    endDate = nextIsoDate(endDate);
+    endUtc = zonedDayStartIso(endDate, timezone);
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-core/src/activity/digest.ts]: Fixed in 6f3680d6a. The aliasing was the `while (endUtc <= startUtc)` end-push in `activityDayWindow`: it stretched the skipped date onto the next real day's interval. Removed it — the naive window for a skipped date is already the zero-width `[transition, transition)`, which is now the documented contract: a civil date that never occurred owns no interval. It contributes to no daily total (`daysOverlappingWeek` already drops zero-width days via `endMs > startMs`) and a connector sync for it fetches nothing (empty SQL day range / empty API date filter); 2011-12-29 and 2011-12-31 each keep their own full windows. Regressions (both red on 978c9b3fb, green on 6f3680d6a): Apia 2011-12-30 → `[2011-12-30T10:00Z, 2011-12-30T10:00Z)` and `notDeepEqual` vs the 31st; weekly summary across the jump attributes each card to exactly one daily total with no `2011-12-30` day entry. Suites: core digest 19/19, weekly 7/7, omi 12/12 + 7/7, reitti 13/13 + 21/21, fireflies 10/10, granola 8/8; `check:pre-push` all gates pass.

File: packages/remnic-core/src/activity/digest.ts [hunk 9]
```diff
@@ -109,9 +109,24 @@ function zonedDayStartIso(date: string, timezone: string): string {
...
+    // binary-search a 48h bracket at minute granularity. The bracket is
+    // safe: `<date>T00:00Z − 24h` is always locally before `date`, and
+    // `<date>T00:00Z + 24h` is always locally on/after it.
+    let lo = Date.parse(`${date}T00:00:00Z`) - 24 * 3_600_000;
+    let hi = lo + 48 * 3_600_000;
+    while (hi - lo > 60_000) {
+      const mid = lo + Math.floor((hi - lo) / 2 / 60_000) * 60_000;
+      if (activityDateInTimezone(new Date(mid), timezone) >= date) hi = mid;
+      else lo = mid;
+    }
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-core/src/activity/digest.ts:128]: Confirmed against the tree and filed as #2814 — reproduced on 6f3680d6a: `activityDayWindow("1844-12-31", "Asia/Manila")` = `[1844-12-31T15:56:00Z, 1844-12-31T15:57:00Z)` (60000 ms wide for a date that never existed; its start instant is locally 1844-12-30). Root cause is the minute-granularity snap in `zonedDayStartIso`'s binary search (`hi - lo > 60_000` with minute-aligned midpoints) plus the `lo = mid` update leaving `hi` on a non-satisfying boundary. Out of this PR's final-round scope by the round cap; tracked in #2814 with the fix direction (sub-minute convergence + boundary invariant + Manila 1844 regressions).
